lessons/codap-exploring/langs/en-us/pages/codap-apply-functions.adoc

```diff
@@ -8,83 +8,79 @@
+
+@n Why do you think the word `Pounds` has different a different color than the rest of the formula? Does `pounds` (lower-case `p`) appear in a special color? @fitb{}{}
 
-@fitb{}{@ifsoln{_`Weight (lbs)` * 3_}}
+@fitb{}{@ifsoln{'Pounds' appeared in pink, because CODAP recognizes that is an attribute in the dataset. Case matters!}}
 
-- Compare your results with the first table, above.
+@n On the second table in @starter-file{expressions-functions, the file}, enter a formula that will increase the number of weeks by 2. Write your formula here: @fitb{}{@ifsoln{`Weeks + 2`}}
 
-@n Edit the `test` column's formula so that 2 weeks are added to the time to adoption for all of the animals. What formula did you use?
+@n Now, let's revisit the @starter-file{animals}. Can you think of other scenarios where it would be useful to enter an expression or function as the formula for a new attribute? Write down one possible scenario. @fitb{}{}
 
-@fitb{}{@ifsoln{_`Time to adoption (weeks)` + 2_}}
+@fitb{}{}
 
 @vspace{1ex}
 
 == Applying Functions
 
-In order to answer each of the questions below, you must first select `test` in @starter-file{animals}, and then choose `Edit formula` from the drop-down menu that appears. You will be typing into the `Edit Formula` box.
+In order to answer each of the questions below, you must first create a new column in @starter-file{animals}. Click on the attribute name for that new column and choose `Edit Formula` from the drop-down menu that appears. You will be typing into the `Edit Formula` box.
 
-- Enter `sqrt(16)` into the `Edit Formula` box.
-
-@n How many arguments does `sqrt` expect? @fitb{}{@ifsoln{_1_}}
+@n Enter `sqrt(16)` into the `Edit Formula` box. How many arguments does `sqrt` expect? @fitb{}{@ifsoln{_1_}}
 
 @n What type of argument does the function expect? (Number? String? Boolean?) @fitb{}{@ifsoln{_Number._}}
 
 @n What type of output does `sqrt` produce? @fitb{}{@ifsoln{_Number._}}
 
 @n What did the function evaluate to? @fitb{}{@ifsoln{_4_}}
 
-- Now, enter `sqrt(Weight)`.
-
-@n What do you observe? @fitb{}{@ifsoln{_CODAP finds the square root of each animal's weight_}}
+@n Now, enter `sqrt(Weight)`. What do you observe? @fitb{}{@ifsoln{_CODAP finds the square root of each animal's weight_}}
 
 @n What happens when you enter `sqrt(Legs)`? @fitb{}{@ifsoln{_CODAP finds the square root of each animal's legs for `sqrt(Legs)`._}}
 
 @n What about `sqrt(Name)`? @fitb{}{@ifsoln{_CODAP produces an error message because it cannot find the square root of a string, only a number._}}
 
-- Enter `stringLength(Name)` into the `test` column.
-
-@n What do you observe? @fitb{}{@ifsoln{__7 appears, CODAP finds the amount of letters in each animal's name__}}
+@n Enter `stringLength(Name)` into the `test` column. @n What do you observe? @fitb{}{@ifsoln{__7 appears, CODAP finds the amount of letters in each animal's name__}}
```

Comment:
I removed a bunch of these questions because codap's functionality conflicts with the appropriate responses. I'm curious whether there is really a big gain from introducing this function... if yes, I think the revised workbook page is openended enough to touch on it without digging into the murky stuff... but, if no, I vote you remove this function from the page.